docs/developer-guide/ksqldb-clients/java-client.md

```diff
@@ -767,6 +768,42 @@ System.out.println(description.name()
 );
 ```
 
+Define variables for substitution<a name="variable-substitution"></a>
+---------------------------------------------------------------
+
+Starting with ksqlDB 0.18.0, users can define session variables by calling the [`define()`](/api/io/confluent/ksql/api/client/Client.html#define(java.lang.String,boolean,java.lang.Object)) method and
+reference them in other functions by wrapping the variable name in `${}`. The [`undefine()`](/api/io/confluent/ksql/api/client/Client.html#undefine(java.lang.String)) method
+undefines a session variable, and [`getVariables()`](/api/io/confluent/ksql/api/client/Client.html#getVariables()) returns a map of the currently defined variables
+and their values. Substitution is supported for the following functions:
+* `streamQuery`
+* `executeQuery`
+* `executeStatement`
+* `describeSource`
+* `createConnector`
+* `dropConnector`
+* `describeConnector`
```

Comment:
Just to confirm: we're planning to add support for `insertInto()` and `streamInserts()` in a future PR, right?

Reply:
Yup, that will be added after substitution is supported for the REST endpoint.